### PR TITLE
Create sets.md

### PR DESCRIPTION
**Pull request by mfa34**
### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [ ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by mfa34** [comment]
@SSwiniarski  i finished. Can you check it and i have a question? When we take a new assignment then are we create a new branch or can we continue old branch ?

**Comment by SSwiniarski** [comment]
> @SSwiniarski i finished. Can you check it and i have a question? When we take a new assignment then are we create a new branch or can we continue old branch ?

@mfa34 You create a new branch off of the main fork.

**Comment by mfa34** [comment]
@SSwiniarski Thanks for your  answer, Sir.

File: content/kotlin/concepts/set/sets.md
Comment (caupolicandiaz) [inline content/kotlin/concepts/set/sets.md:9]:
```suggestion
  - 'Sets'
```

File: content/kotlin/concepts/set/sets.md
Comment (caupolicandiaz) [inline content/kotlin/concepts/set/sets.md:20]:
```suggestion
In Kotlin, the `Set` data type can store elements of various types. For instance, basic data types like `Int`, `String`, `Double`, `Boolean`, as well as complex data types like custom classes, can be used within a `Set`.
```

File: content/kotlin/concepts/set/sets.md
Comment (caupolicandiaz) [inline content/kotlin/concepts/set/sets.md:22]: I think this block with the data types is a bit overkill. I would recommend replacing this section with a basic syntax block before jumping into some of the specific variations that follow.

File: content/kotlin/concepts/set/sets.md
Comment (caupolicandiaz) [inline content/kotlin/concepts/set/sets.md:41]:
```suggestion
Kotlin offers two main types of `Sets`:

```

File: content/kotlin/concepts/set/sets.md
Comment (caupolicandiaz) [inline content/kotlin/concepts/set/sets.md:42]:
```suggestion
### 1. Mutable Set

```

Comment (caupolicandiaz) [reply content/kotlin/concepts/set/sets.md:42]: There should always be an empty line after a header

File: content/kotlin/concepts/set/sets.md
Comment (caupolicandiaz) [inline content/kotlin/concepts/set/sets.md:43]:
```suggestion
A Mutable Set is a dynamic collection that allows you to add, remove, or modify its elements after creation. It is implemented using the `mutableSetOf()` function.
```

Comment (caupolicandiaz) [reply content/kotlin/concepts/set/sets.md:43]: boldface is generally reserved to highlighting the first instance of the subject in the entry. Otherwise, it should be used sparingly to highlight important terms, values, etc

File: content/kotlin/concepts/set/sets.md
Comment (caupolicandiaz) [inline content/kotlin/concepts/set/sets.md:48]:
```suggestion
#### Creating An Empty MutableSet

```kotlin
val emptySet: MutableSet<Int> = mutableSetOf()
```

```

Comment (caupolicandiaz) [reply content/kotlin/concepts/set/sets.md:48]: update the other subsections as above

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@mfa34 I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Review by caupolicandiaz — APPROVED** [review]
@mfa34 There were several changes that were still missing. For the sake of expediency I went ahead and made the changes directly. Please review for future reference. Now, this is ready for a second review.

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:26]:
```suggestion
A mutable set is a dynamic collection that allows the programmer to add, remove, or modify its elements after creation. It is implemented using the `mutableSetOf()` function.
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:101]: This example is wrong.

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:114]:
```suggestion
## Example
```

We also like a sentence before the example summarizing what it is doing.

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:28]:
```suggestion
Creating an empty mutable set:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:34]:
```suggestion
Creating a mutable set:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:40]:
```suggestion
Modifying a mutable set:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:47]:
```suggestion
Parameters of `mutableSetOf()`:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:57]:
```suggestion
Creating an empty immutable set:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:63]:
```suggestion
Creating an immutable set
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:69]:
```suggestion
Parameters of `setOf()`:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:78]:
```suggestion
Adding and removing elements:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:85]:
```suggestion
Checking for a element's existence:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:91]:
```suggestion
Iterating over elements:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:99]:
```suggestion
Retrieving set size:
```

File: content/kotlin/concepts/sets/sets.md
Comment (SSwiniarski) [inline content/kotlin/concepts/sets/sets.md:107]:
```suggestion
Converting a set to a list or array:
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@mfa34 I did a first review and left a few comments.

**Comment by mfa34** [comment]
@SSwiniarski  i updated sir.

**Review by SSwiniarski — APPROVED** [review]
@mfa34 Looks good, but I think @caupolicandiaz needs to look at this again before we merge since this version is so different from the one he approved.

**Review by caupolicandiaz — APPROVED** [review]
@mfa34 @SSwiniarski Looks good. I was considering several of these changes as well, I think it's cleaner without so many headers and some of the consolidation.

**Comment by SSwiniarski** [comment]
@mfa34 Congrats on the new entry. Live link here: https://www.codecademy.com/resources/docs/kotlin/sets

image

**Pull request merged**
